File: DataStructures/Stack.js
class Node {
	constructor({ value, nextNode }) {
		if (!value) {
			throw new Error('missing params in node creation')
		}
		this.value = value
		this.nextNode = nextNode
	}
}

class Stack {
	constructor() {
		this.top = undefined
	}

	printTop() {
		console.log(this.top)
	}

	push(value) {
		const newNode = new Node({ value, nextNode: this.top })
		this.top = newNode
	}

	pop() {
		if (this.top === undefined) {
			return 'empty'
		}

		let removeNode = this.top
		const removeNodeValue = removeNode.value

		this.top = removeNode.nextNode

		removeNode = null
		return removeNodeValue
	}
}

const stack = new Stack()

stack.push(1)
stack.push(2)
stack.push(3)

console.log(stack.pop())
console.log(stack.pop())
console.log(stack.pop())
console.log(stack.pop())

File: DataStructures/Stack.py
class Node:
    value = None
    nextNode = None

    def __init__(self, **nodeValues):
        self.value = nodeValues.get('value')
        self.nextNode = nodeValues.get('nextNode')


class Stack:
    top = None

    def printTop(self):
        print(self.top.value)

    def push(self, value):
        newNode = Node(value=value, nextNode=self.top)
        self.top = newNode

        def pop(self):
            if(self.top == None):
                return 'Empty'

        removedNode = self.top

        removedNodeValue = removedNode.value

        if(removedNode.nextNode):
            self.top = removedNode.nextNode
        else:
            self.top = None

        del removedNode

        return removedNodeValue


newStack = Stack()
newStack.push(1)
newStack.push(2)
newStack.push(3)

print(newStack.pop())
print(newStack.pop())
print(newStack.pop())
print(newStack.pop())
